test/assignment-4/FractionalWrapper.t.sol

```diff
@@ -0,0 +1,128 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.13;
+
+import "forge-std/Test.sol";
+import {FractionalWrapper} from "src/assignment-4/FractionalWrapper.sol";
+import {YearnVaultMock} from "src/assignment-4/YearnVaultMock.sol";
+import {FunToken} from "src/assignment-3/FunToken.sol";
+
+abstract contract ZeroState is Test {
+    
+    FractionalWrapper wrapper;
+    YearnVaultMock yvToken;
+    FunToken token;
+    uint256 constant RAY = 1e27;
+
+    address USER = address(1);
+
+    function setUp() public virtual {
+        token = new FunToken();
+        yvToken = new YearnVaultMock(address(token), "Yearn Vault Mock", "YVMock", token.decimals());
+        yvToken.setPricePerShareMock(2 * RAY);
+        wrapper = new FractionalWrapper(address(token), address(yvToken), "Fractional Wrapper", "FracWrap", yvToken.decimals());
+
+        token.mint(USER, 5000);
+        vm.prank(USER);
+        token.approve(address(wrapper), 5000);
+    }
+}
+
+contract ZeroStateTest is ZeroState {
+    function testDepositingIntoWrapperSendsTokensToYVault() public {
```

Comment:
@alcueca I think this is more of a code style thing. I'm used to writing unit-tests where each test tests for "one thing". For future tests, I can write them so that they include all the things happening in the executing path